command/operator_scheduler_set_config.go

```diff
@@ -0,0 +1,172 @@
+	"github.com/mitchellh/cli"
+	"github.com/posener/complete"
+)
+
+// Ensure OperatorSchedulerSetConfig satisfies the cli.Command interface.
+var _ cli.Command = &OperatorSchedulerSetConfig{}
+
+type OperatorSchedulerSetConfig struct {
+	Meta
+
+	// The scheduler configuration flags allow us to tell whether the user set
+	// a value or not. This means we can safely merge the current configuration
+	// with user supplied, selective updates.
+	schedulerAlgorithm       string
+	memoryOversubscription   flagHelper.BoolValue
+	rejectJobRegistration    flagHelper.BoolValue
+	pauseEvalBroker          flagHelper.BoolValue
+	preemptBatchScheduler    flagHelper.BoolValue
+	preemptServiceScheduler  flagHelper.BoolValue
+	preemptSysBatchScheduler flagHelper.BoolValue
+	preemptSystemScheduler   flagHelper.BoolValue
+}
+
+func (o *OperatorSchedulerSetConfig) AutocompleteFlags() complete.Flags {
+	return mergeAutocompleteFlags(o.Meta.AutocompleteFlags(FlagSetClient),
+		complete.Flags{
+			"-scheduler-algorithm": complete.PredictSet(
+				string(api.SchedulerAlgorithmBinpack),
+				string(api.SchedulerAlgorithmSpread),
+			),
+			"-memory-oversubscription":    complete.PredictSet("true", "false"),
+			"-reject-job-registration":    complete.PredictSet("true", "false"),
+			"-pause-eval-broker":          complete.PredictSet("true", "false"),
+			"-preempt-batch-scheduler":    complete.PredictSet("true", "false"),
+			"-preempt-service-scheduler":  complete.PredictSet("true", "false"),
+			"-preempt-sysbatch-scheduler": complete.PredictSet("true", "false"),
+			"-preempt-system-scheduler":   complete.PredictSet("true", "false"),
+		},
+	)
+}
+
+func (o *OperatorSchedulerSetConfig) AutocompleteArgs() complete.Predictor {
+	return complete.PredictNothing
+}
+
+func (o *OperatorSchedulerSetConfig) Name() string { return "operator scheduler set-config" }
+
+func (o *OperatorSchedulerSetConfig) Run(args []string) int {
+
+	flags := o.Meta.FlagSet("set-config", FlagSetClient)
+	flags.Usage = func() { o.Ui.Output(o.Help()) }
+
+	flags.StringVar(&o.schedulerAlgorithm, "scheduler-algorithm", "", "")
+	flags.Var(&o.memoryOversubscription, "memory-oversubscription", "")
+	flags.Var(&o.rejectJobRegistration, "reject-job-registration", "")
+	flags.Var(&o.pauseEvalBroker, "pause-eval-broker", "")
+	flags.Var(&o.preemptBatchScheduler, "preempt-batch-scheduler", "")
+	flags.Var(&o.preemptServiceScheduler, "preempt-service-scheduler", "")
+	flags.Var(&o.preemptSysBatchScheduler, "preempt-sysbatch-scheduler", "")
+	flags.Var(&o.preemptSystemScheduler, "preempt-system-scheduler", "")
+
+	if err := flags.Parse(args); err != nil {
+		return 1
+	}
+
+	// Set up a client.
+	client, err := o.Meta.Client()
+	if err != nil {
+		o.Ui.Error(fmt.Sprintf("Error initializing client: %s", err))
+		return 1
+	}
+
+	// Fetch the current configuration. This will be used as a base to merge
+	// user configuration onto.
+	resp, _, err := client.Operator().SchedulerGetConfiguration(nil)
```

Comment:
If we add `-check-index` we should fail here if `resp.ModifyIndex != check-index` because that means someone else updated the scheduler config and the user could end up with unintentional results. Imagine during an outage:

1. Operator Alice runs `nomad operator scheduler set-config -reject-job-registration=true`
2. Operator Bob runs `nomad operator scheduler set-config -pause-eval-broker=true`

Without CAS there's no way to know which of the following will be the final state of the scheduler config:

1. RejectJobRegistration=true; PauseEvalBroker=false
2. RejectJobRegistration=false; PauseEvalBroker=true
3. RejectJobRegistration=true; PauseEvalBroker=true

Because the HTTP GETs and PUTs may interleave in any order.

Both Alice and Bob must _manually_ do a Get-before-Set and put the ModifyIndex form the result into the `-check-index` flag to ensure the resulting state is what they intend.